### ISSUE TITLE
when using the 'numbersign' in metrics window, it disappears along with all ligatures made with the numbers sign

### ISSUE THREAD

**Issue by anthony4on6** [fontforge/fontforge#3357]
* Provide a general summary of the issue in the **Title** above.  
* Before you open an issue, please check if a similar issue already exists or has been closed before.

### Important
Mark with [x] to select. Leave as [ ] to unselect.

### When reporting a bug/issue:
- [ ] Screenshot
- [ ] The FontForge version and the operating system you're using
- [ ] The behavior you expect to see, and the actual behavior
- [ ] Steps to reproduce the behavior 
- [ ] \(optional) Possible solution/fix/workaround

### When you open an issue for a change/improvement/feature request:
- [ ] A description of the problem you're trying to solve, including _why_ you think this is a problem
- [ ] If the feature changes current behavior, reasons why your solution is better
- [ ] \(optional) Possible solution/fix/workaround

**Comment by skef** [comment]
I'm a civilian (i.e. not a project member) who has been looking at other parts of the code and took at look at this.

I believe the relevant lines are 426-440 in `fontforgeexe/wordlistparser.c`. A non-escaped numbersign is interpreted as the start of a comment (whereas (for example) `/#21` is interpreted as (hexidecimal) unicode codepoint 0x21). 

However, while there is probably a bug to be fixed here, it's also inadvisable to use that character in any glyph name, including ligatures. You can get that slot to appear with the strings `\u0023` or `/#23` or `/numbersign` (if you use that glyph name). Any glyph name with characters not part of the [Adobe glyph name standard](https://github.com/adobe-type-tools/agl-specification#6-assigning-glyph-names-in-new-fonts) is "asking for it". This is _especially_ true of '`#`', which is the comment character in feature files. 

In short, you may be better off renaming the glyphs that currently include that character with more standard alternatives.

**Comment by anthony4on6** [comment]
Ah, so I was using, as an example, "c_numbersign" for my ligature, but I should use "c_/numbersign" or "c_\u0023" or "c_/#23" instead. 
I am making my own font sets for fingering of wind and brass instruments. # is used for the sharp sign in musical notation. So to show the fingers for c sharp in the 5th octave I would write c#5 to create the fingering for that note using the ligature.
Good to know about the #, but as in retail, the customer is always right, so in programming - the user is always right. - And users are always going to find a use case that hasn't been thought of.
